xm-commons-topic/src/main/java/com/icthh/xm/commons/topic/TopicManager.java

```diff
@@ -0,0 +1,186 @@
+package com.icthh.xm.commons.topic;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
+import com.icthh.xm.commons.topic.config.MessageListenerContainerBuilder;
+import com.icthh.xm.commons.topic.domain.ConsumerHolder;
+import com.icthh.xm.commons.topic.domain.TopicConfig;
+import com.icthh.xm.commons.topic.domain.TopicConsumersSpec;
+import com.icthh.xm.commons.topic.message.MessageHandler;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.time.StopWatch;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.listener.AbstractMessageListenerContainer;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class TopicManager implements RefreshableConfiguration {
+
+    private static final String CONSUMER_CONFIG_PATH_PATTERN = "/config/tenants/{tenant}/{ms}/topic-consumers.yml";
+    private static final String TENANT_NAME = "tenant";
+
+    private AntPathMatcher matcher = new AntPathMatcher();
+    private ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory());
+
+    @Getter
+    private Map<String, Map<String, ConsumerHolder>> topicConsumers = new ConcurrentHashMap<>();
```

Comment:
suggest to rename as `tenantTopicConsumers`

Reply:
+